src/AzureAuth/CommandMain.cs

```diff
@@ -159,6 +159,13 @@ public CommandMain(CommandExecuteEventData eventData, ILogger<CommandMain> logge
         [Option(ConfigOption, "The path to a configuration file.", CommandOptionType.SingleValue)]
         public string ConfigFilePath { get; set; }
 
+        /// <summary>
+        /// Gets or sets the cache file name.
+        /// </summary>
+        [LegalFileName]
+        [Option(CacheFileNameOption, "Override the default cache file location.", CommandOptionType.SingleOrNoValue)]

```

Comment:
I don't think here must have a value. This parameter is from command line, and if it is empty it will be replaced by `msal_{tenant_id}.cache`. Also, it could be replaced by environment variable. If this option is required, all existing command line may not work.

Reply:
We want to use `SingleValue` because providing the option `--cache` by itself has no meaning - and we want to treat that as bad input. i.e. if you are using `--cache` you must give us a value for it.

Reply:
Sorry I thought `SingleValue` means if there is no such option the program won't run.